docs/rules/no-static-typos.md

```diff
@@ -0,0 +1,59 @@
+# Prevent casing typos when declaring static propTypes, contextType and defaultProps
+
+Ensure no casing typos were made declaring static class properties
+
+**Fixable:** This rule is automatically fixable using the `--fix` flag on the command line.
```

Comment:
It's not that it has to just be stylistic - but generally eslint errs on the side of caution to avoid breaking code.

If this rule warns, it'll only be for two reasons: 1) a typo, which is totally autofixable - and 2) a badly but intentionally chosen name, and the user needs to know to manually rename it.